### DISCUSSION TITLE
Leistungsanpassung Wallbox bei gleichzeitigem Betrieb von SAE Geräten

### DISCUSSION THREAD

**Discussion by stevoboy** [camueller/SmartApplianceEnabler#359] [Q&A]
Hallo, 
Diese Frage ist evtl überflüssig, aber interessiert mich vor Inbetriebnahme unserer Wallboxen.

Wenn ich bspw. 4 kWh PV-Leistung verfügbar habe und 2 kWh für die eingeplante WaMa verbrauche, würde der SHM/SAE dann die Wallbox mit 2kwh gleichzeitig einschalten und dynamisch bis auf 4kwh hochregeln wenn die WaMa weniger verbraucht?

**Comment by camueller** [comment]
Ja, genau das mach der SHM. Siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/pics/shm/Verbraucherbilanz_GuterTag.png

**Reply by stevoboy** [reply]
Prima. Muss die Wallbox dann mit niedrigere Prio laufen als die WaMa, damit der SHM das macht?

**Reply by camueller** [reply]
Ich habe die Priorität aller Verbraucher unverändert auf "hoch" belassen. Bei der Wallbox regelt sich das ja darüber, dass diese Überschussbedarf anmeldet, der automatisch eine geringere Priorität hat, als andere Verbraucher.

**Reply by stevoboy** [reply]
Noch eine Frage, seit letzter Woche läuft auch unsere OpenWB Pro.
Irgendwann würde ich mich an die Einbindung in den SAE machen.
Gibt es für OpenWB Wallboxen schon ein Template oder ein Wiki?
Ich würde sie über die http Schnittstelle ansteuern (read/write geht bereits über HomeKit)

Ich erstelle auch gerne eine Anleitung - nur doppelte Arbeit möchte ich nach Möglichkeit vermeiden.

**Reply by camueller** [reply]
Bisher wollten die OpenWB-User ihre Wallbox im SAE nur visualisieren. Insofern existiert bisher kein Template. Wenn Du eins erstellst, werde ich es übernehmen und für alle verfügbar machen.

**Reply by stevoboy** [reply]
Die meisten openwb Nutzer werden wohl nicht die Pro haben und daher die Gerätesteuerung der openwb Standard selbst überlassen + SHM ggf. nur als Zähler.

Da ist mein Usecase eher selten.
Ich würde es aber trotzdem angehen, wenn bei mir das Einschalten des Datenpunkts bei der Wärmepumpe mal nachweislich funktioniert. Vorher macht die Wallbox Integration auch noch keinen sinn.

**Reply by stevoboy** [reply]
@camueller 
Ich habe jetzt mal mal eine Basis Konfiguration angelegt. Bevor ich die Einbindung in das SunnyPortal machen, bitte mal anschauen.
Ich arbeite derzeit noch mit Testdatenpunkten im Iobroker, die ich mit NodeRed Logiken beschalte. Ob direkte Interaktion mit der Wallbox möglich ist, muss ich mir noch überlegen.

Noch eine Frage, den State of Charge habe ich ebenfalls als Datenpunkt im iobroker durch einen VW-Adapter hinterlegt. Da brauche ich eigentlich auch kein Script mehr sondern müsste den Datenpunkt als http Befehl direkt im SAE hinterlegen können.

![image](https://user-images.githubusercontent.com/84937873/214540929-54ccd38e-ca70-4b27-9d56-dc84afc5a461.png)



![image](https://user-images.githubusercontent.com/84937873/214539145-04479d49-73cb-4666-8b6e-c22c2e22ebf4.png)

![image](https://user-images.githubusercontent.com/84937873/214539551-1c690904-a49a-4181-bc53-ee06e3fe97de.png)

![image](https://user-images.githubusercontent.com/84937873/214539938-730dd043-e645-4cf5-b672-779b926276e1.png)

![image](https://user-images.githubusercontent.com/84937873/214540081-e64ce04b-d181-4fb8-9a4e-c77670050e8b.png)

**Comment by camueller** [comment]
Sieht für mich soweit gut aus. Muss halt für Dich funktionieren! Damit es als Vorlage genutzt werden kann, muss der Zugriff direkt auf die Wallbox erfolgen ohne NodeRED.

> Noch eine Frage, den State of Charge habe ich ebenfalls als Datenpunkt im iobroker durch einen VW-Adapter hinterlegt. Da brauche ich eigentlich auch kein Script mehr sondern müsste den Datenpunkt als http Befehl direkt im SAE hinterlegen können.

Doch, der SAE bietet nur einen Script-Aufruf an. Wenn der SOC bei Dir per HTTP kommt, musst Du in dem Script `curl` oder `wget` aufrufen.

**Comment by stevoboy** [comment]
Also wenn ich gleich direkt an die Wallbox gehen soll brauche ich etwas Starthilfe bei der Syntax. Mir ist nämlich nicht klar, was ich im SAE da eingeben müsste.

Mit diesem http request lese ich den Status der Wallbox aus. Daraus kann ich "Ladestecker nicht angeschlossen", "Ladestecker angeschlossen", "Ladevorgang läuft" mit dem richtigen Parametern auslesen. "Ladegerät meldet Fehler" kriege ich wahrscheinlich nicht ausgelesen:
![image](https://user-images.githubusercontent.com/84937873/214938750-e2caee0a-6d40-44d0-b604-e81acb061af6.png)

Bzgl. Schreiben nehme ich aktuell diese zwei Befehle:
"Ladestromstärke setzen" : 
![image](https://user-images.githubusercontent.com/84937873/214939258-de9a7a70-df00-4b36-a8f4-99b5483bd8bc.png)

Zu verwendene Phasen setzen. Das kann der SAE nicht, oder?
![image](https://user-images.githubusercontent.com/84937873/214939494-71e5843e-cd82-44ec-80fa-aabb8b18b3bb.png)

Für "Ladevorgang beginnen" gibt es keinen Befehl bei der OpenWB Pro. "Ladevorgang beenden" löst man so aus:

![image](https://user-images.githubusercontent.com/84937873/214940102-a10dd0f6-079d-4848-bca8-d85f7c12ae12.png)

**Reply by camueller** [reply]
Du kannst Dir einfach ein zweites Browserfenster aufmachen und dort im SAE die Vorlage für den go-eCharger auswählen ohne zu speichern. Dann wird Dir vemutlich schnell klar, wie das aussehen muss. Der go-eCharger liefert auch ein Status-Objekt, in dem auf bestimmte Werte geprüft wird.

Die Phasenanzahl kann der SAE aktuell nicht setzen.

Weil bei Dir `Ladestromstärke setzen` gleichbedeutend mit `Ladevorgang beginnen` ist, musst Du für letzteres irgendeinen anderen HTTP-Befehl (z.B. den gleichen wie zur Status-Abfrage) konfigurieren, der nichts macht. Ansonsten ist die Konfiguration nicht gültig und Du kann sie nicht speichern.

**Reply by stevoboy** [reply]
Ich muss das Thema gerade zurückstellen, weil ich mir noch nicht sicher bin, ob ich den SAE für die Ansteuerung der OpenWB verwenden möchte. Alternativ hatte ich mir überlegt, einfach wieder eine Fritzsteckdose an das Sunny Portal zu hängen. Die OpenWB kann auch mit 2A 1Ph. laden. So könnte ich auch in den Randzeiten mit 0,5 kWh laden und den maximal möglichen Ladestrom per Node Red alle 15 Minuten prüfen - in Abhängigkeit davon, ob gerade eingespeist wird oder nicht. Daher muss ich noch mal schauen, was ich mache. Die Konfiguration des SAEs mit einem Template ist ja auch nicht ganz einfach.

Eine Sache noch, ich betreibe neben dem regulären SMA Wechselrichter und SHM 2.0 noch Hoymiles Modulwechselrichter. Der SHM erkennt die Erzeugung jedoch nicht, sondern reduziert den Gesamtverbrauch um den Erzeugungswert der Hoymiles.
SMA Standardvorgehen wäre ein SMA Energy Meter zwischen Wechselrichter und Zählerschrank. Wäre der SAE alternativ in der Lage einen Energy Meter mit Einspeisung zu simulieren?

![image](https://user-images.githubusercontent.com/84937873/220204175-ae815a09-5061-4f9f-93d8-198f90ce1c07.png)

**Comment by camueller** [comment]
> Wäre der SAE alternativ in der Lage einen Energy Meter mit Einspeisung zu simulieren?

Nein
